### PR TITLE
Scheduler: skip updates of assumed pods

### LINKED ISSUE #100261: Scheduler: skip updates of assumed pods
#### What would you like to be added:

Currently the scheduler may add an assumed pod back to the queue if the update didn't include specific fields (see https://github.com/kubernetes/kubernetes/blob/58814dbe3fc13602fb8a9ac67f9e046c9eb09326/pkg/scheduler/eventhandlers.go#L307).

There doesn't seem to be a reason why we can't simply ignore all updates of assumed pods. Assumed pods are waiting to be bound, if binding succeeded then the pod is scheduled. If binding fails, the pod's latest version will be put back into the unschedulable queue anyways.

#### Why is this needed:

Performance: we save on unnecessary scheduling attempts.

This is a followup on the discussion we had in https://github.com/kubernetes/kubernetes/issues/98828

/sig scheduling
/triage accepted
/help

### PR DESCRIPTION
**Pull request by tanjing2020**
#### What type of PR is this?
/kind feature

#### What this PR does / why we need it:
https://github.com/kubernetes/kubernetes/issues/98828#issuecomment-786190185
1. ignore updates to assumed pods

#### Which issue(s) this PR fixes:

Fixes #100261 

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: pkg/scheduler/eventhandlers.go [hunk 1]
```diff
@@ -299,60 +299,6 @@ func responsibleForPod(pod *v1.Pod, profiles profile.Map) bool {
...
-	// Gets the assumed pod from the cache.
-	assumedPod, err := sched.SchedulerCache.GetPod(pod)
-	if err != nil {
-		utilruntime.HandleError(fmt.Errorf("failed to get assumed pod %s/%s from cache: %v", pod.Namespace, pod.Name, err))
-		return false
-	}
-
-	// Compares the assumed pod in the cache with the pod update. If they are
-	// equal (with certain fields excluded), this pod update will be skipped.
-	f := func(pod *v1.Pod) *v1.Pod {
```
Comment (alculquicondor) [inline pkg/scheduler/eventhandlers.go:327]: I say we should migrate some of these to `isPodUpdated` in `scheduling_queue.go`.

The function is used to determine if an unschedulabe pod should be moved out into backoff or active.

@ahg-g? WDYT?

File: pkg/scheduler/eventhandlers.go [hunk 2]
```diff
@@ -174,7 +174,7 @@ func (sched *Scheduler) updatePodInSchedulingQueue(oldObj, newObj interface{}) {
 	if oldPod.ResourceVersion == newPod.ResourceVersion {
 		return
 	}
-	if sched.skipPodUpdate(newPod) {
+	if isAssumed, _ := sched.SchedulerCache.IsAssumedPod(newPod); isAssumed {
```
Comment (alculquicondor) [inline pkg/scheduler/eventhandlers.go:177]: never discard errors

File: pkg/scheduler/eventhandlers.go (hunk 1 above)
Comment (ahg-g) [reply pkg/scheduler/eventhandlers.go:327]: yes, we can add `ManagedFields` and `Finalizers`. I am not sure about `Annotations`, and `NodeName` isn't going to trigger a queue update.

File: pkg/scheduler/eventhandlers.go (hunk 2 above)
Comment (ahg-g) [reply pkg/scheduler/eventhandlers.go:177]: it seems the new recommendation is to use `utilruntime.HandleError` in this case.

**Comment by alculquicondor** [comment]
@tanjing2020 can you handle the review comments?

**Comment by alculquicondor** [comment]
also, can we consider this for 1.21?

**Comment by tanjing2020** [comment]
Sorry for the late processing, the latest modification has been submitted. /recc @alculquicondor

**Comment by alculquicondor** [comment]
@ahg-g what do you think about qualifying this for 1.21?

File: pkg/scheduler/eventhandlers.go (hunk 1 above)
Comment (alculquicondor) [reply pkg/scheduler/eventhandlers.go]: wait, this comment hasn't been addressed

**Comment by ahg-g** [comment]
we also should remove the one in scheduler.go https://github.com/kubernetes/kubernetes/blob/623d15a8c023fa15b4be9c020dff4460480ec2aa/pkg/scheduler/scheduler.go#L645

**Comment by ahg-g** [comment]
> @ahg-g what do you think about qualifying this for 1.21?

I wish, but I don't think so.

File: pkg/scheduler/eventhandlers.go (hunk 1 above)
Comment (alculquicondor) [reply pkg/scheduler/eventhandlers.go:327]: also `p.Status.Conditions` please

Comment (tanjing2020) [reply pkg/scheduler/eventhandlers.go:327]: p.Status = v1.PodStatus{} has been assigned in isPodUpdated

Comment (alculquicondor) [reply pkg/scheduler/eventhandlers.go:327]: ah missed, that

/lgtm

**Pull request merged**